src/wp-admin/customize.php

```diff
@@ -84,9 +84,17 @@
 	}
 }
 
-$url       = ! empty( $_REQUEST['url'] ) ? sanitize_text_field( $_REQUEST['url'] ) : '';
-$return    = ! empty( $_REQUEST['return'] ) ? sanitize_text_field( $_REQUEST['return'] ) : '';
-$autofocus = ! empty( $_REQUEST['autofocus'] ) ? sanitize_text_field( $_REQUEST['autofocus'] ) : '';
+$url    = ! empty( $_REQUEST['url'] ) ? sanitize_text_field( $_REQUEST['url'] ) : '';
+$return = ! empty( $_REQUEST['return'] ) ? sanitize_text_field( $_REQUEST['return'] ) : '';
+
+/**
+ * Since `$_REQUEST['autofocus'] is an array, we need to sanitize both the keys and values separately.
+ * Merge the sanitized keys & values together using `array_combine`
+ * Set `$autofocus` to the new combined array or an empty array.
+ */
+$autofocus_array_keys   = ! empty( $_REQUEST['autofocus'] ) ? array_map( 'sanitize_key', array_keys( $_REQUEST['autofocus'] ) ) : array(); // Sanitizing the Keys
+$autofocus_array_values = ! empty( $_REQUEST['autofocus'] ) ? array_map( 'sanitize_text_field', array_values( $_REQUEST['autofocus'] ) ) : array(); // Sanitizing the values
+$autofocus              = ( ! empty( $autofocus_array_keys ) && ! empty( $autofocus_array_values ) ) ? array_combine( $autofocus_array_keys, $autofocus_array_values ) : array(); // Combining the keys and values.
```

Comment:
We don't really need to sanitize the keys as `WP_Customize_Manager::set_autofocus()` has an allowlist of keys: `panel`, `section`, `control`

So this should suffice:

```php
$autofocus = ! empty( $_REQUEST['autofocus'] ) && is_array( $_REQUEST['autofocus'] ) ? array_map( 'sanitize_text_field', wp_unslash( $_REQUEST['autofocus'] ) ) : array();
```

Also just noticing the other `wp_unslash` calls further below are incorrect, they should be happening before the sanitizing.

We can thus change the whole code section to this:

```php
$url       = ! empty( $_REQUEST['url'] ) ? sanitize_text_field( wp_unslash( $_REQUEST['url'] ) ) : '';
$return    = ! empty( $_REQUEST['return'] ) ? sanitize_text_field( wp_unslash( $_REQUEST['return'] ) ) : '';
$autofocus = ! empty( $_REQUEST['autofocus'] ) && is_array( $_REQUEST['autofocus'] )
		? array_map( 'sanitize_text_field', wp_unslash( $_REQUEST['autofocus'] ) )
		: array();

if ( ! empty( $url ) ) {
	$wp_customize->set_preview_url( $url );
}
if ( ! empty( $return ) ) {
	$wp_customize->set_return_url( $return );
}
if ( ! empty( $autofocus ) ) {
	$wp_customize->set_autofocus( $autofocus );
}
```

Reply:
Fixed